hs_modelinstance/models.py

```diff
@@ -37,23 +40,40 @@ def remove(cls, element_id):
 
 class ExecutedBy(AbstractMetaDataElement):
     term = 'ExecutedBY'
-    name = models.CharField(max_length=500)
-    url = models.URLField()
+    name = models.CharField(max_length=500, choices=(('-','    '),))
+    model_program_fk = models.ForeignKey('hs_model_program.ModelProgramResource', null=True, blank=True)
+
 
     def __unicode__(self):
         self.name
 
     @classmethod
     def create(cls, **kwargs):
-        return ExecutedBy.objects.create(**kwargs)
+        shortid = kwargs['name']
+
+        obj = get_object_or_404(ModelProgramResource,short_id=shortid)
```

Comment:
You can't raise an Http404 here because you don't know that create was called in response to a web request.